starknet-rpc-test/get_transaction_by_hash.rs

```diff
@@ -0,0 +1,51 @@
+extern crate starknet_rpc_test;
+
+use rstest::rstest;
+use starknet_ff::FieldElement;
+use starknet_providers::Provider;
+use starknet_rpc_test::constants::{ARGENT_CONTRACT_ADDRESS, SIGNER_PRIVATE};
+use starknet_rpc_test::fixtures::madara;
+use starknet_rpc_test::utils::{create_account, AccountActions};
+use starknet_rpc_test::{MadaraClient, Transaction, TransactionResult};
+
+#[rstest]
+#[tokio::test]
+async fn work_valid_transaction_hash(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    let account = create_account(rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS, true);
+    let txs = madara
+        .create_block_with_txs(vec![Transaction::Execution(account.transfer_tokens(
+            FieldElement::from_hex_be("0x123").unwrap(),
+            FieldElement::ONE,
+            None,
+        ))])
+        .await;
+
+    let rpc_response = match txs.unwrap().remove(0).unwrap() {
+        TransactionResult::Execution(rpc_response) => rpc_response,
+        _ => panic!("expected execution result"),
+    };
+
+    // there is a delay between the transaction being available at the client
+    // and the sealing of the block, hence sleeping for 100ms
+    tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
+
+    // not validating the fields inside the transaction as
+    // that is covered in get_block_with_txs
+    assert!(rpc.get_transaction_by_hash(rpc_response.transaction_hash).await.is_ok());
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn fail_invalid_transaction_hash(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    assert!(rpc.get_transaction_by_hash(FieldElement::ZERO).await.is_err());
```

Comment:
cannot we have a more precise match here ?

Reply:
fixed